src/components/implementation/no_interface/cFE_booter/osfilesys.c

```diff
@@ -0,0 +1,326 @@
+** Tar Level Methods
+******************************************************************************/
+
+/*
+ * Loads the position in memory of linked tar file system
+ * Checks for badly linked or no linked tar file.  The names given by
+ * the linker are non-intuative so a description of error checking is given:
+ * First checks to make sure that symbols have been overwritten by linking
+ * process.  Next checks that the size is greater than 0.  Finally checks that
+ * the end of the tar is after the start
+ */
+uint32 tar_load()
+{
+    if (!_binary_cFEfs_tar_start)
+                                        return OS_FS_ERR_DRIVE_NOT_CREATED;
+    if (! &_binary_cFEfs_tar_size)
+                                        return OS_FS_ERR_DRIVE_NOT_CREATED;
+    if (&_binary_cFEfs_tar_end < &_binary_cFEfs_tar_start)
+                                        return OS_FS_ERR_DRIVE_NOT_CREATED;
+    size    = (size_t) &_binary_cFEfs_tar_size;
+    start   = &_binary_cFEfs_tar_start;
+    end     = &_binary_cFEfs_tar_end;
+    return OS_FS_SUCCESS;
+}
+
+/*
+ * parses a loaded tar into fs[0]
+ */
+uint32 tar_parse()
+{
+    assert(start && end);
+    assert(size < INT32_MAX);
+    assert(end - start > 0);
+    assert(size == (size_t) (end - start));
+    assert(fs[0].root->filedes == 1); //first fs should be rooted by first file
+
+    if (!fs[0].devname)                      return OS_FS_ERR_DRIVE_NOT_CREATED;
+
+    struct fsobj *o = fs[0].root;
+    uint32 offset = 0;
+
+    while (offset + start < end) {
+        //tar ends after two empty records
+        if ( !(offset + start)[0] && !(offset + start)[512]) return OS_SUCCESS;
+
+        if (tar_cphdr(offset, o))     return OS_FS_ERR_DRIVE_NOT_CREATED;
+        if (file_insert(o, &fs[0]))   return OS_FS_ERR_DRIVE_NOT_CREATED;
+
+        //data is alligned to 512 byte blocks.  a header is 500 bytes, and
+        // the file's data begins exactly after the header
+        // therefor the next header is 500 + o->size rounded up to a mult of 512
+        offset += round_up(o->size + 500);
+        if (newfile_get(&o))         return OS_FS_ERR_DRIVE_NOT_CREATED;
+    }
+
+    //tar ends before two empty records are found
+    return OS_FS_ERROR;
+}
+
+/*
+ * Copies up to req_sz bytes from the tar file into buf
+ */
+int32 tar_read(uint32 offset, char *buf, uint32 req_sz)
+{
+    if (offset + req_sz > size) return OS_FS_ERROR;
+    if (req_sz == 0)            return OS_FS_SUCCESS;
+    uint32 i;
+    for (i = 0 ; i < req_sz ; i++) {
+        buf[i] = start[i + offset];
+    }
+    return OS_FS_SUCCESS;
+}
+
+/*
+ * Copies information from a tar file header to a fsobj
+ */
+uint32 tar_cphdr(uint32 tar_offset, struct fsobj *file)
+{
+    //TODO: store filenames without paths
+    assert(tar_offset < size);
+    assert(file->filedes > 0);
+
+    char *location = start;
+    location += tar_offset;
+    file -> name        = location;
```

Comment:
Small spacing error on the arrow. Also align the `location = start` equals sign.